site/app/models/CalendarInfo.php

```diff
@@ -119,21 +133,48 @@ public static function loadGradeableCalendarInfo(Core $core, array $gradeables_o
             // Put data of current section into the by-section map
             $info->items_by_sections[] = $curr_section;
         }
+        foreach ($courses as $course) {
+            $cal_items = $calendar_items[$course->getTitle()];
+            foreach ($cal_items as $cal_item) {
+                $date = $cal_item->getDate()->format('Y-m-d');
+                try {
+                    $curItem = [
+                        'title' => htmlspecialchars($cal_item->getText()),
+                        'status' => $cal_item->getTypeString(),
+                        'course' => $course,
+                        'icon' => '',
+                        'url' => '',
+                        'show_due' => false,
+                        'submission' => '',
+                        'status_note' => '',
+                        'color' => $info->colors[$course->getSemester() . $course->getTitle()],
+                        'type' => 'item'
+                    ];
+                    $info->items_by_date[$date][] = $curItem;
+                }
+                catch (\Exception $e) { //Empty catch to skip this item
+                }
```

Comment:
Is there a more specific exception we can catch here?  Why is this try-catch necessary in the first place?

Reply:
So I think that this try-catch is just in case the current calendar item doesn't work for some reason. I really don't think that this try-catch is really necessary but maybe I should make a different way to check if the calendar item is working (although I'm really not sure how to do that without a try-catch). Currently I'm going to delete it but let me know if you want this done differently.

Reply:
Okay so update. I tried to remove it and I get a Parsing Error but the error comes from outside of the try-catch yet somehow doesn't cause problems when the try-catch is in the code. I really have no idea why this is a problem and why the code is working when the try-catch is there. I don't see what this changes. If you have any idea please let me know, I'll take another look with fresh eyes to try to figure it out again.